frontend/src/_helpers/utils.js

```diff
@@ -30,59 +30,78 @@ export function resolve(data, state) {
   }
 }
 
-export function resolveAll(data, state) {
+export function resolveReferences(object, state, defaultValue, customObjects = {}, v2 = false) {
```

Comment:
instead of `v2`, can we name the param to something more readable?

Reply:
ok changed to `withError`